Tests/FluentAssertions.Specs/Numeric/NumericAssertionSpecs.BeNaN.cs

```diff
@@ -0,0 +1,490 @@
+﻿using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Numeric;
+
+public partial class NumericAssertionSpecs
+{
+    public class BeNaN
+    {
+        [Fact]
+        public void When_a_float_value_is_NaN_it_should_succeed()
+        {
+            // Arrange
+            float actual = float.NaN;
+
+            // Act
+            Action act = () => actual.Should().BeNaN();
+
+            // Assert
+            act.Should().NotThrow();
```

Comment:
```suggestion
            // Act / Assert
            actual.Should().BeNaN();
```

We now combine assertions that are not supposed to throw into one.

(And following)